src/js_native_api_v8.cc

```diff
@@ -767,32 +692,29 @@ uint32_t Reference::Ref() {
   if (persistent_.IsEmpty()) {
     return 0;
   }
-  uint32_t refcount = RefBase::Ref();
-  if (refcount == 1 && can_be_weak_) {
+  if (++refcount_ == 1 && can_be_weak_) {
     persistent_.ClearWeak();
   }
-  return refcount;
+  return refcount_;
 }
 
 uint32_t Reference::Unref() {
   // When the persistent_ is cleared in the WeakCallback, and a second pass
   // callback is pending, return 0 unconditionally.
-  if (persistent_.IsEmpty()) {
+  if (persistent_.IsEmpty() || refcount_ == 0) {
```

Comment:
This seems like a logic change. Do we have a unit test covering this behaviour?

Reply:
It is the same logic as before. The `RefBase` used to have this code:
```C++
uint32_t RefBase::Unref() {
  if (refcount_ == 0) {
    return 0;
  }
  return --refcount_;
}
```
When I expanded it in place, it caused adding the new condition above to return 0, and the use of the `old_refcount` became redundant as it was protecting from the `RefBase::Unref` returning 0 when the ref count is already 0.